### PR TITLE
Ndlano/issues#1709 learning path search

### PR DESCRIPTION
**Pull request by talepre**
Connects to NDLANO/Issues#1709

### CODE REVIEW

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 1]
```diff
@@ -42,26 +47,24 @@ class AddResourceModal extends Component {
...
+      }
       this.setState({ selected });
     } else {
       this.setState({ selected: {}, article: {} });
     }
-  }
+  };
 
-  async onPaste(e) {
+  onPaste = async e => {
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: evt

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 2]
```diff
@@ -86,21 +89,50 @@ class AddResourceModal extends Component {
...
-      return result;
+      const result =
+        this.props.type === 'urn:resourcetype:learningPath'
+          ? await this.searchLearningpath(input)
+          : await this.groupSearch(input);
+      return result.map(current => ({
+        ...current,
+        title: current.title ? current.title.title : '',
+      }));
     } catch (e) {
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: err

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 3]
```diff
@@ -86,21 +89,50 @@ class AddResourceModal extends Component {
...
+          verificationStatus: 'CREATED_BY_NDLA',
+        }
+      : {
+          pageSize: 10,
+          language: 'nb',
+          fallback: true,
+          verificationStatus: 'CREATED_BY_NDLA',
+        };
+    const res = await learningpathSearch(query);
+    return res.results ? res.results : [];
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: `return res.results || []`;

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 4]
```diff
@@ -115,18 +147,40 @@ class AddResourceModal extends Component {
...
     if (selected.id) {
       try {
         this.setState({ loading: true });
+        const resourceId =
+          this.props.type === 'urn:resourcetype:learningPath'
+            ? await this.findResourceIdLearningPath(selected)
+            : getResourceIdFromPath(selected.paths[0]);
         await createTopicResource({
-          resourceId: getResourceIdFromPath(selected.paths[0]),
+          resourceId: resourceId,
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: `resourceId,`

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 5]
```diff
@@ -115,18 +147,40 @@ class AddResourceModal extends Component {
...
           topicid: topicId,
         });
+        if (topicFilters.length > 0) {
+          await createResourceFilter({
+            filterId: topicFilters[0].id,
+            relevanceId: topicFilters[0].relevanceId,
+            resourceId: resourceId,
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: `resourceId,`

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 6]
```diff
@@ -136,7 +190,31 @@ class AddResourceModal extends Component {
...
+    } else {
+      try {
+        await createResource({
+          contentUri: `urn:learningpath:${learningpath.id}`,
+          name: learningpath.title,
+        });
+        const resource = await queryLearningPathResource(learningpath.id);
+        const resourceId = resource[0].id;
+        await createResourceResourceType({
+          resourceId: resourceId,
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: `resourceId,`

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 7]
```diff
@@ -136,7 +190,31 @@ class AddResourceModal extends Component {
...
+          resourceTypeId: 'urn:resourcetype:learningPath',
+        });
+        return resource[0].id;
+      } catch (e) {
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: err

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 8]
```diff
@@ -188,6 +266,7 @@ AddResourceModal.propTypes = {
   type: PropTypes.string,
   allowPaste: PropTypes.bool,
   topicId: PropTypes.string.isRequired,
+  topicFilters: PropTypes.array,
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: arrayOf

File: src/containers/StructurePage/resourceComponents/ResourceGroup.jsx [hunk 9]
```diff
@@ -109,7 +110,7 @@ ResourceGroup.propTypes = {
   activeFilter: PropTypes.string,
   locale: PropTypes.string,
   currentTopic: PropTypes.shape({
-    filter: PropTypes.array,
+    filters: PropTypes.array,
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/ResourceGroup.jsx]: arrayOf filtershape

File: src/modules/learningpath/learningpathApi.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+import {
+  resolveJsonOrRejectWithError,
+  apiResourceUrl,
+  fetchAuthorized,
+} from '../../util/apiHelpers';
+
+const baseUrl = apiResourceUrl('/learningpath-api/v2/learningpaths');
+
+export const learningpathSearch = async query =>
```
Comment (chrpeter) [inline src/modules/learningpath/learningpathApi.js]: async await

File: src/modules/taxonomy/resources/index.js [hunk 11]
```diff
@@ -112,3 +112,11 @@ export function queryResources(articleId, language) {
     )}&?language=${language}`,
   ).then(resolveJsonOrRejectWithError);
 }
+
+export function queryLearningPathResource(learningpathId) {
```
Comment (chrpeter) [inline src/modules/taxonomy/resources/index.js]: async await

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 12]
```diff
@@ -136,7 +190,31 @@ class AddResourceModal extends Component {
...
+      try {
+        await createResource({
+          contentUri: `urn:learningpath:${learningpath.id}`,
+          name: learningpath.title,
+        });
+        const resource = await queryLearningPathResource(learningpath.id);
+        const resourceId = resource[0].id;
+        await createResourceResourceType({
+          resourceId: resourceId,
+          resourceTypeId: 'urn:resourcetype:learningPath',
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: Finnes i constants.js

File: src/containers/StructurePage/resourceComponents/AddResourceModal.jsx [hunk 13]
```diff
@@ -115,18 +147,40 @@ class AddResourceModal extends Component {
...
-  async addSelected() {
-    const { topicId, refreshResources, onClose } = this.props;
+  addSelected = async () => {
+    const { topicId, refreshResources, onClose, topicFilters } = this.props;
     const { selected } = this.state;
     if (selected.id) {
       try {
         this.setState({ loading: true });
+        const resourceId =
+          this.props.type === 'urn:resourcetype:learningPath'
```
Comment (chrpeter) [inline src/containers/StructurePage/resourceComponents/AddResourceModal.jsx]: finnes i constants

File: src/modules/taxonomy/resources/index.js (hunk 11 above)
Comment (talepre) [reply src/modules/taxonomy/resources/index.js]: det tror jeg ikke at jeg trenger her?

Comment (chrpeter) [reply src/modules/taxonomy/resources/index.js]: Tja 
```
const result = await fetchAuthorized(
    `${baseUrl}/queries/resources/?contentURI=${encodeURIComponent(
      `urn:learningpath:${learningpathId}`,
    )}`,
  );
return  resolveJsonOrRejectWithError(result);
```

Comment (chrpeter) [reply src/modules/taxonomy/resources/index.js]: Men trenger ikke nei

**Pull request merged**